src/main/java/com/j256/ormlite/dao/DaoManager.java

```diff
@@ -213,57 +213,43 @@ public static synchronized void registerDaoWithTableConfig(ConnectionSource conn
 	/**
 	 * Clear out all of internal caches.
 	 */
-	public static synchronized void clearCache() {
-		if (configMap != null) {
-			configMap.clear();
-			configMap = null;
-		}
+	public static void clearCache() {
+		configMap.clear();
 		clearDaoCache();
 	}
 
 	/**
 	 * Clear out our DAO caches.
 	 */
-	public static synchronized void clearDaoCache() {
-		if (classMap != null) {
-			classMap.clear();
-			classMap = null;
-		}
-		if (tableConfigMap != null) {
-			tableConfigMap.clear();
-			tableConfigMap = null;
-		}
+	public static void clearDaoCache() {
+		classMap.clear();
+		tableConfigMap.clear();
 	}
 
 	/**
 	 * This adds database table configurations to the internal cache which can be used to speed up DAO construction.
 	 * This is especially true of Android and other mobile platforms.
 	 */
-	public static synchronized void addCachedDatabaseConfigs(Collection<DatabaseTableConfig<?>> configs) {
-		Map<Class<?>, DatabaseTableConfig<?>> newMap;
-		if (configMap == null) {
-			newMap = new HashMap<Class<?>, DatabaseTableConfig<?>>();
-		} else {
-			newMap = new HashMap<Class<?>, DatabaseTableConfig<?>>(configMap);
-		}
+	public static void addCachedDatabaseConfigs(Collection<DatabaseTableConfig<?>> configs) {
+		Map<Class<?>, DatabaseTableConfig<?>> newMap = new HashMap<Class<?>, DatabaseTableConfig<?>>(configMap);
 		for (DatabaseTableConfig<?> config : configs) {
 			newMap.put(config.getDataClass(), config);
 			logger.info("Loaded configuration for {}", config.getDataClass());
 		}
-		configMap = newMap;
+		configMap.putAll(newMap);
 	}
 
-	private static void addDaoToClassMap(ClassConnectionSource key, Dao<?, ?> dao) {
-		if (classMap == null) {
-			classMap = new HashMap<ClassConnectionSource, Dao<?, ?>>();
+	private static Dao<?, ?> addDaoToClassMap(ClassConnectionSource key, Dao<?, ?> dao) {
+		Dao<?, ?> old = classMap.putIfAbsent(key, dao);
+		if (old != null) {
+			return old;
+		} else {
+			return dao;
 		}
```

Comment:
```suggestion
		if (old != null) {
			return old;
		}
		
		return dao;
```

Reply:
Actually I like the returns at the same level in this case.  What I would ration is if old == null return dao else return old.

Reply:
Ahh, I follow the "bail out early" design pattern, explained here: https://medium.com/swlh/return-early-pattern-3d18a41bba8

Reply:
Oh, ok. So since it's j256's project, I'll redo with returns on the same level, alright?